node-notes-cli/app.js

```diff
@@ -0,0 +1,57 @@
+import { readFile, writeFile } from 'node:fs/promises';
+
+let data = await readFile('data.json', 'utf8');
+data = JSON.parse(data);
+const notesObj = data.notes;
+
+async function rewrite() {
+  await writeFile('data.json', JSON.stringify(data, null, 2));
+}
+
+async function read() {
+  for (const [key, value] of Object.entries(notesObj)) {
+    console.log(`${key}: ${value}`);
+  }
+}
+
+async function add() {
+  const id = data.nextId;
+  data.nextId++;
```

Comment:
FYI, this also works and would be easily read and understood by most developers:
`const id = data.nextId++;`